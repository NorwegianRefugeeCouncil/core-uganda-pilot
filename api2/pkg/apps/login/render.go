package login

import "net/http"

func (s *Server) Render(w http.ResponseWriter, req *http.Request, templateName string, data map[string]interface{}) {
	if err := s.template.ExecuteTemplate(w, templateName, data); err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
}
