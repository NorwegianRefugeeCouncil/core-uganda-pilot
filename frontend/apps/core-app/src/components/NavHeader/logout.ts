import { Platform } from 'react-native';
import * as AuthSession from 'expo-auth-session';
import { openAuthSessionAsync } from 'expo-web-browser';
import Constants from 'expo-constants';

const authorizationEndpoint = `${Constants.manifest?.extra?.issuer}/oauth2/sessions/logout`;
const redirectUri = AuthSession.makeRedirectUri({ useProxy: false });

export const logout = async () => {
  try {
    await openAuthSessionAsync(
      `${authorizationEndpoint}?post_logout_redirect=${redirectUri}`,
      'redirectUrl',
    );
    if (Platform.OS === 'web') {
      sessionStorage.removeItem('tokenResponse');
    }
  } catch (err) {
    console.error(err);
  }
};
