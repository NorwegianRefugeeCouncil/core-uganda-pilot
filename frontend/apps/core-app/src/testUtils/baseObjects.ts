import { FormDefinition, FormType, Record } from 'core-api-client';

export const baseForm: FormDefinition = {
  id: 'form1',
  code: '',
  name: 'form 1',
  databaseId: 'databaseId1',
  folderId: 'folderId1',
  formType: FormType.DefaultFormType,
  fields: [],
};

export const baseTextField = {
  id: 'field1',
  name: 'field 1',
  description: 'description 1',
  code: '',
  required: false,
  key: false,
  fieldType: {
    text: {},
  },
};

export const baseMultilineTextField = {
  id: 'field2',
  name: 'field 2',
  description: 'description 2',
  code: '',
  required: true,
  key: true,
  fieldType: {
    multilineText: {},
  },
};

export const baseRecord1: Record = {
  id: 'record1',
  values: [
    {
      value: 'text',
      fieldId: 'field1',
    },
    {
      value: 'multi line text',
      fieldId: 'field2',
    },
  ],
  formId: 'form1',
  databaseId: 'databaseId1',
  ownerId: undefined,
};
