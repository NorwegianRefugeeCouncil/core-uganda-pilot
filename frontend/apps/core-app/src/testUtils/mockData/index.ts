export { makeRecord } from './Record';
export { makeForm } from './Form';
export { makeField } from './FormField';
