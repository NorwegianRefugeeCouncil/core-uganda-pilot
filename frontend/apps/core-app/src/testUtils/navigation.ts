export const mockNavigationProp = {
  dispatch: jest.fn(),
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getParent: jest.fn(),
  getState: jest.fn(),
  setParams: jest.fn(),
  setOptions: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
};
