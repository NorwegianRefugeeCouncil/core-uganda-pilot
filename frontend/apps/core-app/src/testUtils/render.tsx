import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import {
  render as r,
  RenderAPI,
  RenderOptions,
} from '@testing-library/react-native';
import { render as rWeb } from '@testing-library/react';
import { NativeBaseProvider } from 'native-base';
import { theme } from 'core-design-system';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

export const render = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
): RenderAPI =>
  r(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );

export const renderWeb = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
) =>
  rWeb(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );
