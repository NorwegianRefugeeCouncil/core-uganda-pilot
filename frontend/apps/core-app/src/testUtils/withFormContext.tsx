import { Button } from 'native-base';
import React from 'react';
import { FieldValues, FormProvider, useForm } from 'react-hook-form';

export const withFormContext =
  <T,>(
    Component: React.FC<T>,
    defaultValues: FieldValues = {},
    onSubmit: (data: any) => void = () => {},
  ) =>
  // eslint-disable-next-line react/display-name
  (props: T) => {
    const f = useForm({ defaultValues });

    const handleSubmit = f.handleSubmit((data: any) => {
      onSubmit(data);
    });

    return (
      <FormProvider {...f}>
        <Component {...props} />
        <Button testID="with-form-context-submit-button" onPress={handleSubmit}>
          Submit
        </Button>
      </FormProvider>
    );
  };
