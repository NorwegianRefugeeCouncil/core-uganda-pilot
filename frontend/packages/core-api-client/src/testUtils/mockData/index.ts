export * from './Form';
export * from './FormField';
export * from './Record';
