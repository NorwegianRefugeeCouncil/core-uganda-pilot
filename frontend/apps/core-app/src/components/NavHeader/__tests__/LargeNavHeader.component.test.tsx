import { fireEvent } from '@testing-library/react-native';
import { Linking } from 'react-native';

import { render } from '../../../testUtils/render';
import { LargeNavHeaderComponent } from '../LargeNavHeader.component';
import * as logout from '../logout';

it('should match the snapshot', () => {
  const { toJSON } = render(
    <LargeNavHeaderComponent
      route={{
        key: '',
        name: '',
        params: {},
      }}
    />,
  );

  expect(toJSON()).toMatchSnapshot();
});

it('should render the root route header', () => {
  const { getByTestId } = render(
    <LargeNavHeaderComponent
      route={{
        key: '',
        name: '',
        params: {},
      }}
    />,
  );

  expect(getByTestId('nav-header')).toHaveTextContent('Beneficiaries');
});

it('should render the recipients list route header', () => {
  const { getByTestId } = render(
    <LargeNavHeaderComponent
      route={{
        key: '',
        name: 'recipientsRoot',
        params: { screen: 'recipientsList' },
      }}
    />,
  );

  expect(getByTestId('nav-header')).toHaveTextContent('Beneficiaries');
});

it('should render the recipients registration route header', () => {
  const { getByTestId } = render(
    <LargeNavHeaderComponent
      route={{
        key: '',
        name: 'recipientsRoot',
        params: { screen: 'recipientsRegistration' },
      }}
    />,
  );

  expect(getByTestId('nav-header')).toHaveTextContent(
    'Beneficiary Registration',
  );
});

it('should call the logout function', () => {
  jest
    .spyOn(Linking, 'openURL')
    .mockImplementationOnce(() => Promise.resolve());

  const logoutSpy = jest.spyOn(logout, 'logout').mockImplementation(() => {});

  const { getByText } = render(
    <LargeNavHeaderComponent
      route={{
        key: '',
        name: 'recipientsRoot',
        params: { screen: 'recipientsList' },
      }}
    />,
  );

  const logoutButton = getByText('Log out');
  fireEvent.press(logoutButton);

  expect(logoutSpy).toHaveBeenCalled();
});
