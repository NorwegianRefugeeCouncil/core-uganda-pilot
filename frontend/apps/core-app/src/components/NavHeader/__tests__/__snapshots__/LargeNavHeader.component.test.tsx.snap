// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": "auto",
        "maxWidth": 1180,
        "paddingBottom": 40,
        "paddingTop": 40,
        "width": "100%",
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <Image
        dataSet={Object {}}
        onError={[Function]}
        source={1}
        style={
          Object {
            "height": 65,
            "maxWidth": "100%",
            "width": 65,
          }
        }
      />
      <View
        dataSet={Object {}}
        style={Object {}}
      />
      <View
        dataSet={Object {}}
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityRole="link"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "height": "auto",
              "marginRight": 32,
              "width": "auto",
            }
          }
        >
          <Text
            dataSet={Object {}}
            level="1"
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#24303E",
                "fontFamily": "Roboto_500Medium",
                "fontSize": 18,
                "fontStyle": undefined,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 26,
                "textAlign": "auto",
                "textDecorationLine": "none",
              }
            }
          >
            Beneficiaries
          </Text>
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="link"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "height": "auto",
              "marginRight": 32,
              "width": "auto",
            }
          }
        >
          <Text
            dataSet={Object {}}
            level="1"
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#24303E",
                "fontFamily": "Roboto_500Medium",
                "fontSize": 18,
                "fontStyle": undefined,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 26,
                "textAlign": "auto",
                "textDecorationLine": "none",
              }
            }
          >
            Register Beneficiary
          </Text>
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="link"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "height": "auto",
              "marginRight": 32,
              "width": "auto",
            }
          }
        >
          <Text
            dataSet={Object {}}
            level="1"
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#24303E",
                "fontFamily": "Roboto_500Medium",
                "fontSize": 18,
                "fontStyle": undefined,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 26,
                "textAlign": "auto",
                "textDecorationLine": "none",
              }
            }
          >
            Settings
          </Text>
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="link"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "height": "auto",
              "marginRight": 32,
              "width": "auto",
            }
          }
        >
          <Text
            dataSet={Object {}}
            level="1"
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#24303E",
                "fontFamily": "Roboto_500Medium",
                "fontSize": 18,
                "fontStyle": undefined,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 26,
                "textAlign": "auto",
                "textDecorationLine": "none",
              }
            }
          >
            Log out
          </Text>
        </View>
      </View>
    </View>
    <View
      dataSet={Object {}}
      style={Object {}}
    />
    <Text
      dataSet={Object {}}
      level="1"
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#000000",
          "fontFamily": "Roboto_700Bold",
          "fontSize": 44,
          "fontStyle": undefined,
          "fontWeight": undefined,
          "letterSpacing": 0,
          "lineHeight": 52,
          "marginTop": 48,
          "textAlign": "auto",
          "textDecorationLine": undefined,
        }
      }
      testID="nav-header"
    >
      Beneficiaries
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
